#VRML_SIM R2020a utf8
WorldInfo {
  gravity 0 0 0
}
Viewpoint {
  fieldOfView 0.8
  orientation 0.39831662618180996 0.8756620069090683 0.27305661494115696 0.20629892573368191
  position 8.45618044381667 0.8196243676611206 49.18175780700454
  lensFlare LensFlare {
  }
}
TexturedBackground {
  texture "dusk"
}
TexturedBackgroundLight {
  luminosity 0.2
}
Robot {
  rotation -1 0 0 -0.1308996938995747
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        position -1.57
        axis 0 0 1
        minStop -1.6
        maxStop 1.6
      }
      device [
        PositionSensor {
          name "sun_sensor"
        }
        RotationalMotor {
          name "sun_motor"
          maxTorque 1000
        }
      ]
      endPoint Solid {
        translation 17.999994292773025 0.014333880793200748 0
        rotation 0 0 -1 1.57
        children [
          PointLight {
            attenuation 0 0 1
            color 1 1 0.7
            intensity 50
            location 0 0 1
            castShadows TRUE
          }
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.8 0.8 0
                emissiveColor 0.8 0.8 0
                shininess 1
                transparency 0.5
              }
            }
            geometry Sphere {
            }
          }
        ]
        name "solid(1)"
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -1.57
        axis 0 0 1
        minStop -1.6
        maxStop 1.4
        dampingConstant 0.2
      }
      device [
        PositionSensor {
          name "panel_sensor"
        }
        RotationalMotor {
          name "motor"
        }
      ]
      endPoint Solid {
        translation 1.999999365863692 0.001592653421466769 0
        rotation 0 0 -1 1.57
        children [
          Solid {
            translation 0 2 0
            children [
              DEF PANEL Shape {
                appearance PBRAppearance {
                  baseColor 0.5 0.5 0.5
                  roughness 0.1
                  metalness 0.8
                }
                geometry Box {
                  size 2 0.1 2
                }
              }
            ]
            name "PANEL"
          }
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.2 0.2 0.2
              }
            }
            geometry DEF HOLDER Cylinder {
              height 4
              radius 0.2
            }
          }
        ]
        boundingObject Cylinder {
          height 0.1
          radius 0.2
        }
        physics Physics {
          mass 0.25
        }
        locked TRUE
      }
    }
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.2 0.2 0.2
        }
      }
      geometry DEF BASE Box {
        size 4 0.4 4
      }
    }
  ]
  controller "solar_pd"
}
